### ISSUE TITLE
Docker container fails to run the model and swagger reports network failure

### ISSUE THREAD

**Issue by linliD** [patrickloeber/docker-fastapi-ml#2]
Hi Patrick,

I was testing if this docker container is actually working, however whenever I finished build, the swagger UI reports "failed to fetch. possible reasons: cors network failure url scheme must be "http" or "https" for cors request."
[This line ](https://github.com/patrickloeber/docker-fastapi-ml/blob/b20e577d4580d5328d94f35493c96bd8dbb9cfcc/model.py#L14C1-L15C1) seems to be the one to cause trouble. 

Any idea why this is the case and how to solve? Thanks in advance.

**Comment by SpiliakosGeorge** [comment]
Exact same issue for me.
Any further info on how to resolve this @linliD?
API works fine until dockerized.

**Comment by patrickloeber** [comment]
Hi, maybe this thread helps? https://stackoverflow.com/questions/73704026/failed-to-fetch-possible-reasons-cors-network-failure-url-scheme-must-be-http

**Comment by SpiliakosGeorge** [comment]
Hey Patrick,

Yes i did try this as well as some other options from here https://github.com/tiangolo/fastapi/issues/1663
I used chrome browser and postman to access the endpoint.
Unfortunately nothing worked and i think the swagger message is not indicating the real problem.
The app gets stuck while performing the forward pass outputs = model(**encoding) and the really interesting fact is that the inference works properly when the API is **NOT** dockerized. 
Thanks

**Comment by SpiliakosGeorge** [comment]
Posting this here in case someone needs this in the future.

https://stackoverflow.com/questions/77290003/segmentation-fault-when-using-sentencetransformer-inside-docker-container

Apparently when your video was uploaded the torch version was 2.0.1 or less and the requirements.txt do not provide the verbose versions of each module.

tldr; use torch 2.0.1 for silicon m1 + docker
